src/inspectors/precompiles/assertion_adopter.rs

```diff
@@ -13,9 +13,74 @@ pub fn get_assertion_adopter(context: &PhEvmContext) -> Result<Bytes, Infallible
 
 #[cfg(test)]
 mod test {
-    use crate::test_utils::run_precompile_test;
+    use super::*;
+    use crate::{
+        inspectors::{
+            phevm::{
+                LogsAndTraces,
+                PhEvmContext,
+            },
+            tracer::CallTracer,
+        },
+        test_utils::{
+            random_address,
+            run_precompile_test,
+        },
+    };
+    use alloy_primitives::Address;
+    use alloy_sol_types::SolValue;
+
+    fn with_adopter_context<F, R>(adopter: Address, f: F) -> R
+    where
+        F: FnOnce(&PhEvmContext) -> R,
+    {
+        let call_tracer = CallTracer::new();
+        let logs_and_traces = LogsAndTraces {
+            tx_logs: &[],
+            call_traces: &call_tracer,
+        };
+
+        let context = PhEvmContext {
+            logs_and_traces: &logs_and_traces,
+            adopter,
+        };
+        f(&context)
+    }
+
+    #[test]
+    fn test_get_assertion_adopter_zero_address() {
+        let adopter = Address::ZERO;
+
+        let result = with_adopter_context(adopter, get_assertion_adopter);
+        assert!(result.is_ok());
+
+        let encoded = result.unwrap();
+        assert!(!encoded.is_empty());
+
+        // Verify we can decode the result back to the original address
+        let decoded = Address::abi_decode(&encoded, false);
+        assert!(decoded.is_ok());
+        assert_eq!(decoded.unwrap(), adopter);
+    }
+
+    #[test]
+    fn test_get_assertion_adopter_random_address() {
```

Comment:
nit: `test_get_assertion_adopter_zero_address()` and `test_get_assertion_adopter_random_address()` have literally the same execution path. Redundancy can be removed by creating a helper function